vector-im/element-android
# Support element call widget (PSG-627)

## Type of change


## Content


We implement Element Call as a widget with some extra functionalities.

## Motivation and context
Since this is a call widget we need to make it run in the background, provide a picture-in-picture mode and also preserve the current native WebRTC call implementation for rooms which does not have an active Element Call widget.

This PR is a cherry-pick version of [PR #6464](https://github.com/vector-im/element-android/pull/6464) which additionally implements a Bluetooth connection with push-to-talk devices. Since it is an experimental PR with a hardcoded device name and service address without a good UX, we wanted to create this PR to be able to merge into develop.



## Screenshots / GIFs

https://user-images.githubusercontent.com/1254401/180235359-19dcf0c3-05a8-4202-8ce3-7359de96a630.mp4







## Tests



- Request an invite (from me?) to the room which has an active Element Call widget
- Click voice call button and this should navigate you to the widget
- Choose a DM room without EC widget and you should be able to make a regular voice call with the same button

## Tested devices

- OS version(s):

## Checklist
